### ISSUE TITLE
"CA2214: Do not call overridable methods in constructors" is not reported for properties

### ISSUE THREAD

**Issue by AlexanderSher** [dotnet/roslyn-analyzers#5044]
### Analyzer

**Diagnostic ID**: [CA2214](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2214): `Do not call overridable methods in constructors" is not reported for properties`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug
CA2214 is not reported for virtual properties because property access is a `OperationKind.PropertyReference`, not a `OperationKind.Invocation`: https://github.com/dotnet/roslyn-analyzers/blob/a0522afd52b25ad211298e308fd5bbc7734865e0/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotCallOverridableMethodsInConstructors.cs#L56

### Steps To Reproduce

```cs
[Fact]
public async Task CA2214VirtualPropertyCSharp()
{
    await VerifyCS.VerifyAnalyzerAsync(@"
class C
{
    C()
    {
        SomeProperty = 5;
    }

    protected virtual int SomeProperty { get; set; }
}
",
        GetCA2214CSharpResultAt(6, 9));
}
```

**Comment by Evangelink** [comment]
I think that's a valuable addition.

**Comment by BrightLight** [comment]
I'm pretty certain that earlier implementations of CA2214 (might be the old FxCop) did report on properties as well, because we have a `[SuppressMessage]` for that diagnostic on some of our constructors exactly because of that (which, funny enough, are now flagged by [IDE0079](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0079))

**Comment by Evangelink** [comment]
I would add that according to the [specs](https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/override), the `override` keyword could be applied to method, property, indexer, or event so IMO all these cases must be handled.

**Comment by Evangelink** [comment]
@mavasani @jmarolf What's your opinion?

**Comment by jmarolf** [comment]
Well considering the rule is "Do not call overridable _methods_ in constructors" I can understand why properties were not considered when porting this over. I do agree that the same risk remains when calling an overridable property as when calling a method, but I would expect the risk to be much lower. 

Considering CA2214 is disabled by default today I am ok changing it behavior to include properties. This change will break people that have turned the rule on though, so I would say with goes in `Warning Level` 7 aka main.

If we want to be conservative, we could make a separate rule for this, and I could see some merits in that. Will need to investigate the old FxCop behavior. If FxCop included properties (and based on the way its analysis works, I would expect it to) then adding this as an addendum in .NET 7 seems perfectly reasonable.

**Comment by Evangelink** [comment]
@jmarolf You can assign me on this task. I am just waiting for your input as to whether I simply extend current rule or add a new one AND whether I target main or something else.

**Comment by Evangelink** [comment]
@jmarolf Have you had time to dig into the old FxCop?
